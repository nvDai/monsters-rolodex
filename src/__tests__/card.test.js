import React from 'react'

import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Card from '../components/card/card.component'

describe('Card', () => {
  it('should render without crashing', () => {
    const mockProps = {
      monster: {
        id: '1',
        name: 'Michael',
        email: 'michael@gmail.com'
      }
    }
    const wrapper = shallow(<Card { ...mockProps } />)

    expect(mockProps.monster.id).toBeTruthy()
    expect(mockProps.monster.name).toBeTruthy()
    expect(mockProps.monster.email).toBeTruthy()
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})