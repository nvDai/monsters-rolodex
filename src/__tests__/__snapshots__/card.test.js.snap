// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render without crashing 1`] = `
<div
  className="card-container"
>
  <img
    alt="Michael"
    src="https://robohash.org/1?set=set2&size=180x180"
  />
  <h1>
    Michael
  </h1>
  <p>
    michael@gmail.com
  </p>
</div>
`;
